responses/embed.go

```diff
@@ -0,0 +1,123 @@
+package responses
+
+import (
+	"github.com/bwmarrin/discordgo"
+)
+
+type EmbedMessage struct {
+	embed *discordgo.MessageEmbed
+}
+
+func (em *EmbedMessage) ToDiscordMessage() *discordgo.MessageSend {
+	return &discordgo.MessageSend{Embed: em.embed}
+}
+
+func NewEmbedMessage(e *discordgo.MessageEmbed) *EmbedMessage {
+	return &EmbedMessage{embed: e}
+}
+
+type Embed struct {
+	*discordgo.MessageEmbed
+}
+
+func NewEmbed() *Embed {
+	return &Embed{&discordgo.MessageEmbed{}}
+}
+
+func (e *Embed) SetTitle(title string) *Embed {
+	e.Title = title
+	return e
+}
+
+func (e *Embed) SetDescription(description string) *Embed {
+	e.Description = description
+	return e
+}
+
+func (e *Embed) AddField(name, value string) *Embed {
+	e.Fields = append(e.Fields, &discordgo.MessageEmbedField{
+		Name:  name,
+		Value: value,
+	})
+
+	return e
+}
+
+func (e *Embed) SetImage(args ...string) *Embed {
+	var URL string
+	var proxyURL string
+
+	if len(args) == 0 {
+		return e
+	}
+	if len(args) > 0 {
+		URL = args[0]
+	}
```

Comment:
전체적으로 이런 패턴이 많이 보이는데, 새로 함수를 구현해도 좋을 것 같습니다.
(아래는 예시)
```suggestion
	if len(args) == 0 {
		return e
	}
	URL := getArgsOrDefault(args, 0, "")
```

Reply:
해당 부분은 그리 여러 번 반복되는 부분이 아니고, 더 많은 부분에서 사용될 코드도 아니라고 생각합니다. 그래서 새로 함수를 구현한다고 더 깔끔해지거나 짧아지진 않을 것 같고, 오히려 관련 함수들의 기능을 한번에 이해하기 위해 다른 곳을 또 읽어줘야하는 불편함이 생긴다고 생각합니다.

Reply:
setImage, setThumbnail, setAuthor 에서 같은 패턴이 반복되고 있어서 반복되는건 꽤 많다고 생각하긴 했고 변수선언, if문으로 나뉘어져있는 코드를 바로 선언하면서 값을 설정할 수 있으면 코드는 깔끔해질것 같아요.
그리고 함수를 구현했을때 naming만 잘한다면 굳이 함수내용을 안 읽어봐도 알 수 있을거고 읽는다고 하더라도 함수가 2줄정도 분량이라 큰 부담이 될것 같진 않아요